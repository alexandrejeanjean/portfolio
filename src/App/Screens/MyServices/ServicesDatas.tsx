import {
  Coffee0,
  Help0,
  Integration0,
  Development0,
  Project0,
} from "../../Assets/imgPath";

interface Service {
  legend: string[];
  img: any;
}

const ServicesDatas: Service[] = [
  { legend: ["Découverte", "du projet"], img: Coffee0 },
  { legend: ["Accompagnement", "& réflexion"], img: Help0 },
  { legend: ["Intégration", "des maquettes"], img: Integration0 },
  { legend: ["Développement", "front-end"], img: Development0 },
  { legend: ["Suivi de projet", "& SAV"], img: Project0 },
];

interface Presentation {
  page: number;
  datas: string[];
}

const ServicePresentationDatas: Presentation[] = [
  {
    page: 0,
    datas: [
      "J'ai pour habitude de travailler en étroite collaboration avec l'UI/UX designer, afin de m'imprégner des guidelines et de les respecter durant la phase d'intégration.",
      'Ecrire du code "éco-friendly", afin de créer des applications maintenables à long terme est l\'un de mes objectifs principal.',
      "J'aime partager, échanger autour des bonnes pratiques, et apprendre d'autrui.",
    ],
  },
  {
    page: 1,
    datas: [
      "Voici la stack technique avec laquelle j'ai pour habitude de travailler :",
      "Html - Scss - Javascript / Typescript",
      "React - React-Native - Redux - Redux-thunk",
      "Jest - Enzyme - Cypress.io",
      "Heroku - Android Studio - Xcode - Git - Gitflow - Jira - Trello",
      "Je développe actuellement les compétences backend suivantes :",
      "Node.js - Express - Sequelize - Chaï - Mocha - Swagger",
      "Ainsi que des notions d'intégration continue :",
      "Docker - Docker compose - Travis",
    ],
  },
  {
    page: 2,
    datas: [
      "En dehors du web, je suis un jeune papa, qui aime observer son fils grandir et profiter de sa famille.",
      "Passionné par le fonctionnement de l'Homme et de la nature, je m'intéresse notamment aux médecines alternatives, médecine traditionnelle chinoise, qi gong, yoga...",
      "Je suis ouvert, flexible et enthousiaste, toujours prêt à découvrir de nouvelle perspectives !",
    ],
  },
];

export { ServicesDatas, ServicePresentationDatas };
