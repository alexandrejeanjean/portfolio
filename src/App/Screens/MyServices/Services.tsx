import React, { useState } from "react";
import ClassicButton from "../../Shared/Button/Button";
import NextPrevButton from "../../Shared/NextPrevBtn/NextPrevBtn";
import { ServicesDatas, ServicePresentationDatas } from "./ServicesDatas";
import "./services.scss";

const Services = () => {
  const [page, setPage] = useState(0);

  return (
    <>
      <main className="page">
        <div className="services-wrapper">
          <section className="services-presentation-wrapper">
            <h1 className="services-title">Mes comp√©tences.</h1>
            {ServicePresentationDatas[page].datas.map((sentence) => (
              <p className="services-subtitle" key={sentence.substr(0, 5)}>
                {sentence}
              </p>
            ))}
            <div>
              <NextPrevButton text="Intro" handleClick={() => setPage(0)} /> //{" "}
              <NextPrevButton
                text="Hard skills"
                handleClick={() => setPage(1)}
              />{" "}
              //{" "}
              <NextPrevButton
                text="Soft skills"
                handleClick={() => setPage(2)}
              />{" "}
            </div>
            <div>
              <ClassicButton text="Visiter les projets" endpoint="/projets" />
            </div>
          </section>

          <section className="skills-wrapper">
            {ServicesDatas.map((item) => (
              <div className="skill-wrapper" key={item.img + "_skill"}>
                <div className="skill-cube">
                  <img src={item.img} alt="" title="" />
                </div>
                <p>
                  {item.legend[0]}
                  <br />
                  {item.legend[1]}
                </p>
              </div>
            ))}
          </section>
        </div>
      </main>
    </>
  );
};

export default Services;
