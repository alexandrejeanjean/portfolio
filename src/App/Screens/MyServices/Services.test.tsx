import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import Services from "./Services";
import { ServicesDatas } from "./ServicesDatas";

let wrapper: any;

beforeEach(() => {
  wrapper = shallow(<Services />);
});

describe("Services view", () => {
  it("should render Services component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should be possible to click button to access next page", () => {
    const component = mount(
      <Router>
        <Services />
      </Router>
    );
    component.find("a.classic-btn").simulate("keydown", { keyCode: 32 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should render 5 skills", () => {
    const component = mount(
      <Router>
        <Services />
      </Router>
    );
    expect(component.find("section.skills-wrapper").children()).toHaveLength(
      ServicesDatas.length
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
