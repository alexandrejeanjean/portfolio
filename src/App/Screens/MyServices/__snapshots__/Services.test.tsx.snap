// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services view should be possible to click button to access next page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Services>
      <main
        className="page"
      >
        <div
          className="services-wrapper"
        >
          <section
            className="services-presentation-wrapper"
          >
            <h1
              className="services-title"
            >
              Mes compétences.
            </h1>
            <p
              className="services-subtitle"
            >
              J'ai pour habitude de travailler en étroite collaboration avec l'UI/UX designer, afin de m'imprégner des guidelines et de les respecter durant la phase d'intégration.
            </p>
            <p
              className="services-subtitle"
            >
              Ecrire du code "éco-friendly", afin de créer des applications maintenables à long terme est l'un de mes objectifs principal.
            </p>
            <p
              className="services-subtitle"
            >
              J'aime partager, échanger autour des bonnes pratiques, et apprendre d'autrui.
            </p>
            <div>
              <NextPrevButton
                handleClick={[Function]}
                text="Intro"
              >
                <button
                  className="next-prev-btn"
                  onClick={[Function]}
                >
                  <p>
                    Intro
                  </p>
                </button>
              </NextPrevButton>
               -
               
              <NextPrevButton
                handleClick={[Function]}
                text="Hard skills"
              >
                <button
                  className="next-prev-btn"
                  onClick={[Function]}
                >
                  <p>
                    Hard skills
                  </p>
                </button>
              </NextPrevButton>
               
              -
               
              <NextPrevButton
                handleClick={[Function]}
                text="Soft skills"
              >
                <button
                  className="next-prev-btn"
                  onClick={[Function]}
                >
                  <p>
                    Soft skills
                  </p>
                </button>
              </NextPrevButton>
               
            </div>
            <div>
              <ClassicButton
                endpoint="/projets"
                text="Visiter les projets"
              >
                <Link
                  className="classic-btn "
                  to="/projets"
                >
                  <LinkAnchor
                    className="classic-btn "
                    href="/projets"
                    navigate={[Function]}
                  >
                    <a
                      className="classic-btn "
                      href="/projets"
                      onClick={[Function]}
                    >
                      Visiter les projets
                    </a>
                  </LinkAnchor>
                </Link>
              </ClassicButton>
            </div>
          </section>
          <section
            className="skills-wrapper"
          >
            <div
              className="skill-wrapper"
              key="function SvgCoffee0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCoffee0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Découverte
                <br />
                du projet
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgHelp0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgHelp0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Accompagnement
                <br />
                & réflexion
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgIntegration0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgIntegration0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Intégration
                <br />
                des maquettes
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgDevelopment0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgDevelopment0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Développement
                <br />
                front-end
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgProject0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgProject0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Suivi de projet
                <br />
                & SAV
              </p>
            </div>
          </section>
        </div>
      </main>
    </Services>
  </Router>
</BrowserRouter>
`;

exports[`Services view should render 5 skills 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Services>
      <main
        className="page"
      >
        <section
          className="services-wrapper"
        >
          <h1
            className="services-title"
          >
            Mes compétences.
          </h1>
          <p
            className="services-subtitle"
          >
            J'ai à coeur d'architecturer, et d'intégrer les différents composants d'une application avec soin.
          </p>
          <p
            className="services-subtitle"
          >
            Je prends le temps de respecter les guidelines des UI/UX Designers.
          </p>
          <p
            className="services-subtitle"
          >
            Délivrer une application maintenable, dev-friendly est mon objectif.
          </p>
          <ClassicButton
            endpoint="/projets"
            text="Visiter les projets"
          >
            <Link
              className="classic-btn "
              to="/projets"
            >
              <LinkAnchor
                className="classic-btn "
                href="/projets"
                navigate={[Function]}
              >
                <a
                  className="classic-btn "
                  href="/projets"
                  onClick={[Function]}
                >
                  Visiter les projets
                </a>
              </LinkAnchor>
            </Link>
          </ClassicButton>
          <div
            className="skills-wrapper"
          >
            <div
              className="skill-wrapper"
              key="function SvgCoffee0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCoffee0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Découverte
                <br />
                du projet
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgHelp0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgHelp0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Accompagnement
                <br />
                & réflexion
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgIntegration0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgIntegration0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Intégration
                <br />
                des maquettes
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgDevelopment0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgDevelopment0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Développement
                <br />
                front-end
              </p>
            </div>
            <div
              className="skill-wrapper"
              key="function SvgProject0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgProject0.name})
  );
}"
            >
              <div
                className="skill-cube"
              >
                <img
                  alt=""
                  src={[Function]}
                  title=""
                />
              </div>
              <p>
                Suivi de projet
                <br />
                & SAV
              </p>
            </div>
          </div>
        </section>
      </main>
    </Services>
  </Router>
</BrowserRouter>
`;

exports[`Services view should render Services component correctly 1`] = `
<Fragment>
  <main
    className="page"
  >
    <div
      className="services-wrapper"
    >
      <section
        className="services-presentation-wrapper"
      >
        <h1
          className="services-title"
        >
          Mes compétences.
        </h1>
        <p
          className="services-subtitle"
        >
          J'ai pour habitude de travailler en étroite collaboration avec l'UI/UX designer, afin de m'imprégner des guidelines et de les respecter durant la phase d'intégration.
        </p>
        <p
          className="services-subtitle"
        >
          Ecrire du code "éco-friendly", afin de créer des applications maintenables à long terme est l'un de mes objectifs principal.
        </p>
        <p
          className="services-subtitle"
        >
          J'aime partager, échanger autour des bonnes pratiques, et apprendre d'autrui.
        </p>
        <div>
          <NextPrevButton
            handleClick={[Function]}
            text="Intro"
          />
           -
           
          <NextPrevButton
            handleClick={[Function]}
            text="Hard skills"
          />
           
          -
           
          <NextPrevButton
            handleClick={[Function]}
            text="Soft skills"
          />
           
        </div>
        <div>
          <ClassicButton
            endpoint="/projets"
            text="Visiter les projets"
          />
        </div>
      </section>
      <section
        className="skills-wrapper"
      >
        <div
          className="skill-wrapper"
          key="function SvgCoffee0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCoffee0.name})
  );
}"
        >
          <div
            className="skill-cube"
          >
            <img
              alt=""
              src={[Function]}
              title=""
            />
          </div>
          <p>
            Découverte
            <br />
            du projet
          </p>
        </div>
        <div
          className="skill-wrapper"
          key="function SvgHelp0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgHelp0.name})
  );
}"
        >
          <div
            className="skill-cube"
          >
            <img
              alt=""
              src={[Function]}
              title=""
            />
          </div>
          <p>
            Accompagnement
            <br />
            & réflexion
          </p>
        </div>
        <div
          className="skill-wrapper"
          key="function SvgIntegration0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgIntegration0.name})
  );
}"
        >
          <div
            className="skill-cube"
          >
            <img
              alt=""
              src={[Function]}
              title=""
            />
          </div>
          <p>
            Intégration
            <br />
            des maquettes
          </p>
        </div>
        <div
          className="skill-wrapper"
          key="function SvgDevelopment0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgDevelopment0.name})
  );
}"
        >
          <div
            className="skill-cube"
          >
            <img
              alt=""
              src={[Function]}
              title=""
            />
          </div>
          <p>
            Développement
            <br />
            front-end
          </p>
        </div>
        <div
          className="skill-wrapper"
          key="function SvgProject0(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgProject0.name})
  );
}"
        >
          <div
            className="skill-cube"
          >
            <img
              alt=""
              src={[Function]}
              title=""
            />
          </div>
          <p>
            Suivi de projet
            <br />
            & SAV
          </p>
        </div>
      </section>
    </div>
  </main>
</Fragment>
`;
